server/tso/allocator_manager.go

```diff
@@ -416,12 +421,28 @@ func longSleep(ctx context.Context, waitStep time.Duration) bool {
 	}
 }
 
-func (am *AllocatorManager) campaignAllocatorLeader(loopCtx context.Context, allocator *LocalTSOAllocator, dcLocationInfo *pdpb.GetDCLocationInfoResponse) {
+func (am *AllocatorManager) campaignAllocatorLeader(loopCtx context.Context, allocator *LocalTSOAllocator, dcLocationInfo *pdpb.GetDCLocationInfoResponse, isNextLeader bool) {
 	log.Info("start to campaign local tso allocator leader",
 		zap.String("dc-location", allocator.dcLocation),
 		zap.Any("dc-location-info", dcLocationInfo),
 		zap.String("name", am.member.Member().Name))
-	if err := allocator.CampaignAllocatorLeader(defaultAllocatorLeaderLease); err != nil {
+	cmps := make([]clientv3.Cmp, 0)
+	nextLeaderKey := path.Join(am.rootPath, allocator.dcLocation, "next-leader")
```

Comment:
It's better to extract this line as a function, so it can be used in `getNextLeaderID`, `deleteNextLeaderID` and here.